src/interactiverecipes.jl

```diff
@@ -0,0 +1,20 @@
+@widget wdg function interactivepartition(t; throttle = 0.1)
+    :x =  @nodeps dropdown(colnames(t), label = "x")
+    :y =  @nodeps dropdown(vcat(Symbol(""), colnames(t)), label = "y")
+    :nparts =  @nodeps slider(1:200, value = 100, label = "number of partitions")
+    :nparts_throttle = Observables.throttle(throttle, :nparts)
+    :dropmissing = @nodeps toggle(false, label = "dropmissing")
+    :by = @nodeps dropdown(colnames(t), multiple = true)
```

Comment:
This `@nodeps` macro is kind of an sore... They all appear as an  `:sym = @nodeps ...` assignment here. I was wondering if it'd be OK to take syntax inspiration/precedent from PlotRecipes and use `:sym <-- slider(...)` to mean `:sym = @nodeps slider(...)`.

I almost want to be convinced that what I just said is a terrible idea.

Reply:
I would actually like to remove syntactic magic as much as possible. One possible way forward is https://github.com/piever/Widgets.jl/pull/25: add definitions in Widgets which would be overloaded by a backend (InteractBase for now, hopefully also Makie in the future). Then `slider` would be defined in Widgets so just `sym = slider(1:100)` would work. I was planning to try and finish up that PR in the near future.

Reply:
I would actually like to remove syntactic magic as much as possible. One possible way forward is https://github.com/piever/Widgets.jl/pull/25: add definitions in Widgets which would be overloaded by a backend (InteractBase for now, hopefully also Makie in the future). Then `slider` would be defined in Widgets so just `sym = slider(1:100)` would work. I was planning to try and finish up that PR in the near future.